content/productivity/repeat-attack/plugin.js

```diff
@@ -0,0 +1,184 @@
+const { default: Manager } = await import(
+  "https://plugins.zkga.me/utils/RepeatAttackCore.js"
+);
+
+const { html, render, useState, useLayoutEffect } = await import(
+  "https://unpkg.com/htm/preact/standalone.module.js"
+);
+
+let Spacing = {
+  marginLeft: "12px",
+  marginRight: "12px",
+};
+let VerticalSpacing = {
+  marginBottom: "12px",
+};
+let Clickable = {
+  cursor: "pointer",
+  textDecoration: "underline",
+};
+let ActionEntry = {
+  marginBottom: "10px",
+  display: "flex",
+  justifyContent: "space-between",
+  color: "",
+};
+
+function centerPlanet(id) {
+  ui.centerLocationId(id);
+}
+
+function Attack({ action }) {
+  return html`
+    <div key=${action.id} style=${ActionEntry}>
+      <span>
+        <span
+          style=${{ ...Spacing, ...Clickable }}
+          onClick=${() => centerPlanet(action.payload.srcId)}
+          >${action.payload.srcId.substring(5, 10)}</span
+        >
+        →
+        <span
+          style=${{ ...Spacing, ...Clickable }}
+          onClick=${() => centerPlanet(action.payload.syncId)}
+          >${action.payload.syncId.substring(5, 10)}</span
+        ></span
+      >
+      <button
+        onClick=${() => {
+          op.delete(action.id);
+        }}
+      >
+        ✕
+      </button>
+    </div>
+  `;
+}
+
+function AddAttack() {
+  let [planet, setPlanet] = useState(ui.getSelectedPlanet());
+  let [source, setSource] = useState(false);
+  let [target, setTarget] = useState(false);
+  let onClick = () => {
+    setPlanet(ui.getSelectedPlanet());
+  };
+  useLayoutEffect(() => {
+    window.addEventListener("click", onClick);
+    return () => {
+      window.removeEventListener("click", onClick);
+    };
+  }, [onClick]);
```

Comment:
As a code suggestion 😄 
```suggestion
  useLayoutEffect(() => {
    let onClick = () => {
      setPlanet(ui.getSelectedPlanet());
    };
    window.addEventListener("click", onClick);
    return () => {
      window.removeEventListener("click", onClick);
    };
  }, []);
```